### DISCUSSION TITLE
socketio.exceptions.BadNamespaceError: / is not a connected namespace.

### DISCUSSION THREAD

**Discussion by leolo0626** [miguelgrinberg/python-socketio#1012] [Q&A]
My code is very similar to the old issue.
https://github.com/miguelgrinberg/Flask-SocketIO/issues/1485
I establish the connection with namespaces before I emit the message.
However, this cause the above error occasionally. And I watched a comment under another old issue. 
https://github.com/miguelgrinberg/python-socketio/issues/634
The repo owner @miguelgrinberg mentions 
> What I've been telling people as a workaround is to move your emit to the connect event handler, and then it'll work **100%**. But for some people this is weird, they prefer to have a sequential logic in their client, so I've seen code that does a sleep(1) between the connect and the emit and that also sort of works.

Could you tell me what is /how to move emit to connect event handler and provide me some examples ? @miguelgrinberg **_Many thanks_**

**Comment by miguelgrinberg** [comment]
This workaround is not necessary anymore. What version of the package are you using?

**Reply by leolo0626** [reply]
python-socket==5.3.0
python-engineio==4.2.0

**Comment by leolo0626** [comment]
I initialize three emit event to the same namespace sequentially . 
Each function will try to check the connection of socket by using `sio.connected`. if connection is okay, then it will not re-connect. Otherwise, we will connect to server again. In addition,  I observed that when first emit occurs, the server side gives the following error. It makes everything fail afterwards.

What is possible reason for the occurrence of this error.

**message async handler error**
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/engineio/asyncio_server.py", line 447, in _trigger_event
    ret = await self.handlers[event](*args)
  File "/usr/local/lib/python3.8/site-packages/socketio/asyncio_server.py", line 565, in _handle_eio_message
    await self._handle_connect(eio_sid, pkt.namespace, pkt.data)
  File "/usr/local/lib/python3.8/site-packages/socketio/asyncio_server.py", line 441, in _handle_connect
    sid = self.manager.connect(eio_sid, namespace)
  File "/usr/local/lib/python3.8/site-packages/socketio/base_manager.py", line 54, in connect
    self.enter_room(sid, namespace, None, eio_sid=eio_sid)
  File "/usr/local/lib/python3.8/site-packages/socketio/base_manager.py", line 121, in enter_room
    self.rooms[namespace][room][sid] = eio_sid
  File "/usr/local/lib/python3.8/site-packages/bidict/_mut.py", line 78, in __setitem__
    self._put(key, val, self.on_dup)
  File "/usr/local/lib/python3.8/site-packages/bidict/_base.py", line 209, in _put
    dedup_result = self._dedup_item(key, val, on_dup)
  File "/usr/local/lib/python3.8/site-packages/bidict/_base.py", line 258, in _dedup_item
    raise ValueDuplicationError(val)
bidict.ValueDuplicationError: xO__Yb4BdGreH-74AAKa

**Reply by miguelgrinberg** [reply]
I'm sorry I cannot follow what you are describing. Please provide an example client and server with all the code so that I can reproduce the problem(s) that you have here.
